app/src/main/java/tech/ula/ui/SessionEditFragment.kt

```diff
@@ -111,9 +117,10 @@ class SessionEditFragment : Fragment() {
                     else -> {
                         // TODO adapter to associate filesystem structure with list items?
                         val filesystem = filesystemList.find { it.name == filesystemName }
-                        session.filesystemName = filesystem!!.name
-                        session.filesystemId = filesystem.id
-                        text_input_username.setText(filesystem.defaultUsername)
+                        filesystem?.run {
```

Comment:
Will you change this to `filesysem?.let {` and `this` to `it` to remain consistent with how things have been done elsewhere in the code?

`run` is a slightly more specific use case than is needed here. 
[some standard library explanations](https://proandroiddev.com/the-tldr-on-kotlins-let-apply-also-with-and-run-functions-6253f06d152b)
[a stack overflow description of receivers](https://stackoverflow.com/questions/45875491/what-is-a-receiver-in-kotlin)

Reply:
Yeah I can change it to let.  Thanks for providing some helpful links